app/src/main/java/ch/epfl/qedit/backend/database/Util.java

```diff
@@ -149,39 +188,24 @@ public static MatrixFormat extractMatrixFormat(Map<String, Object> doc) throws F
 
     public static MatrixFormat.Field extractField(Map<String, Object> field)
             throws FormatException {
-        String typeString = (String) field.get("type");
-        String text = (String) field.get("text");
-        Integer maxCharacters = (Integer) field.get("max_characters");
+        String typeString = (String) field.get(MatrixFormat.Field.TO_MAP_TYPE);
+        String text = (String) field.get(TO_MAP_TEXT);
 
-        if (typeString == null || text == null || maxCharacters == null)
+        if (typeString == null || text == null)
             throw new FormatException(
                     "Invalid field for matrix format: missing type, text or max_characters");
 
-        return new MatrixFormat.Field(extractFieldType(typeString), maxCharacters, text);
-    }
-
-    public static MatrixFormat.Field.Type extractFieldType(String type) throws FormatException {
-        switch (type) {
-            case "pre_filled":
-                return MatrixFormat.Field.Type.PreFilled;
-            case "text":
-                return MatrixFormat.Field.Type.Text;
-            case "unsigned_int":
-                return MatrixFormat.Field.Type.UnsignedInt;
-            case "signed_int":
-                return MatrixFormat.Field.Type.SignedInt;
-            case "unsigned_float":
-                return MatrixFormat.Field.Type.UnsignedFloat;
-            case "signed_float":
-                return MatrixFormat.Field.Type.SignedFloat;
-            default:
-                throw new FormatException("Unknown field type");
+        try {
+            MatrixFormat.Field.Type type = MatrixFormat.Field.Type.valueOf(typeString);
```

Comment:
That's really cool, I didn't forgot about `valueOf()` and `name()`, so I am glad you could use them ;)